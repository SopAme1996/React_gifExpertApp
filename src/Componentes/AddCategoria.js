import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategoria = ({ setData }) => {
  const [input, setInput] = useState("");

  const eventoTeclado = (e) => {
    setInput(e.target.value);
  };

  const eventoEnter = (e) => {
    e.preventDefault();

    if (input.trim().length > 2) {
      setData((data) => [input, ...data]);
      setInput("");
    }
  };

  return (
    <form onSubmit={eventoEnter}>
      <input
        type="text"
        value={input}
        onChange={eventoTeclado}
        placeholder="Ingrese su personaje favorito"
      />
    </form>
  );
};

AddCategoria.propTypes = {
  setData: PropTypes.func.isRequired,
};
