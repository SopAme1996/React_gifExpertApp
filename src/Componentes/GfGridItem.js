import React from "react";
import PropTypes from "prop-types";

export const GfGridItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GfGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GfGridItem;
