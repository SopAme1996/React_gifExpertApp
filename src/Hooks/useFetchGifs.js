import { useState, useEffect } from 'react';
import { getEndPoint } from '../helpers/getGifs';

export const useFetchGifs = (categoria) =>{
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(()=>{
        getEndPoint(categoria)
        .then(img =>{
            setState({
                data:img,
                loading: false
            });
        })
    }, [ categoria ])

    return state; //{data:[], loading: true};
}