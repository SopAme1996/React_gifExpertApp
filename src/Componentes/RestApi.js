import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { GfGridItem } from './GfGridItem';
import PropTypes from 'prop-types'


export const RestApi = ({ categoria }) => {

    const {data:imagenes, loading}= useFetchGifs(categoria);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn titulo'>{categoria}</h3>
        <hr className="line"/>
        <div className="animate__animated animate__backInLeft card-grid">

        {loading && <div className='containerLoading'><p>Loading....</p><div className='laoding'></div></div>}
        {
            imagenes.map(img => {
                return(
               <GfGridItem
                   key={img.id}
                   {...img}
              />
              )
            })
                }
        </div>
        </>
    )
}

RestApi.propTypes = {
categoria: PropTypes.string.isRequired
}