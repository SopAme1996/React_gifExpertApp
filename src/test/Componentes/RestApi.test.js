import React from 'react';
import { RestApi } from '../../Componentes/RestApi';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs');


describe('Hacemos el test del RestApi ', () => {
    const categoria = "Goku";

    test('se debe mostrar correctamente ', () => {
      
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<RestApi categoria={categoria}/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrar items cuando se carga imagenes useFetchGifs', () => {
        const imgs = [{
            id: 'ABC',
            url: 'https://localhost',
            title: 'Cualquier Cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });

        const wrapper = shallow(<RestApi categoria={categoria}/>);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GfGridItem').length).toBe(imgs.length);
    })
    
});