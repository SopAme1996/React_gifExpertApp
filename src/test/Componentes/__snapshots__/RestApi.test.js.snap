// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hacemos el test del RestApi  se debe mostrar correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn titulo"
  >
    Goku
  </h3>
  <hr
    className="line"
  />
  <div
    className="animate__animated animate__backInLeft card-grid"
  >
    <div
      className="containerLoading"
    >
      <p>
        Loading....
      </p>
      <div
        className="laoding"
      />
    </div>
  </div>
</Fragment>
`;
